### DISCUSSION TITLE
How do I get type checking for mdx stories?

### DISCUSSION THREAD

**Discussion by eloisetaylor5693** [storybookjs/storybook#20795] [Help]
If I use a component and give the wrong arguments, I get an error about there being incorrect arguments either in the IDE intellisense or when running the typescript compiler.  When I write an mdx story using invalid props, I don't get any errors when running storybook, just when I load the story in storybook.  

We use typescript for writing our React components.  We have at least 3 stories that have missing props which when selecting in Storybook, we just see an error stack trace.

![image](https://user-images.githubusercontent.com/20887184/214825270-af25bca2-8091-47b6-80bb-30eaf176df73.png)

I would like to be able to see a failure somewhere when I accidentally break the contract of the component

Example component:
```typescript
  interface MyNiceProps {
      hello: string;
  }

  const MyNiceComponent = ({hello} : MyNiceProps) => {
    return (
      <p>{hello}</p>
    );
  };
```

Example mdx
```mdx
import { Canvas, Meta, Story } from '@storybook/addon-docs';
import { MyNiceComponent } from './MyNiceComponent';

<Meta title="MDX/MyNiceComponent" component={MyNiceComponent} />

# Prop typo

<Story name="prop typo"
  <MyNiceComponent sfhfh="hello" />
</Story>
```

How do I get type checking for mdx stories?

**Comment by aditi-gawli** [comment]
Same question , I get an type error saying "Type AccordionProps is not assignable to type Partial<Args> " I am passing mock-data.ts file variable to mdx args
